onnxscript/rewriter/pattern.py

```diff
@@ -563,10 +558,15 @@ def __init__(
         self.rel_tol = rel_tol
         self.abs_tol = abs_tol
 
-    def match_scalar(self, scalar_value, return_value: Sequence[ir.Node]):
-        if math.isclose(scalar_value, self.value, rel_tol=self.rel_tol, abs_tol=self.abs_tol):
-            return MatchResult(return_value)
-        return MatchResult.FAIL()
+    def match_scalar(self, scalar_value):
+        status = math.isclose(
+            scalar_value, self.value, rel_tol=self.rel_tol, abs_tol=self.abs_tol
+        )
+        # Note: If the value is produced by a Constant node, we could include
+        # the Constant node in the return_value list. However, we don't do that.
+        # Instead, we will rely on DCE to remove the constant node if it is not
+        # used elsewhere.
+        return MatchResult(status)
```

Comment:
```suggestion
        return MatchResult(success=status)
```

I somehow feel like the arg in MatchResult would be not so obvious if we don't specify it's success.